rsqueakvm/storage_contexts.py

```diff
@@ -472,35 +495,70 @@ def stacksize(self):
 
     def full_stacksize(self):
         if not self.pure_is_block_context():
-            # Magic numbers... Takes care of cases where reflective
-            # code writes more than actual stack size
             method = self.w_method()
             assert isinstance(method, W_CompiledMethod), "context method is not a W_CompiledMethod"
-            stacksize = method.compute_frame_size()
+            stacksize = method.frame_size()
         else:
-            # TODO why not use method.compute_frame_size for BlockContext too?
             stacksize = self.stacksize()  # no temps
         return stacksize
 
     def init_temps_and_stack(self):
         self = fresh_virtualizable(self)
         stacksize = self.full_stacksize()
-        self._temps_and_stack = [None] * stacksize
+        self._temps_and_stack = [self.space.w_nil] * stacksize
```

Comment:
With the dynamic frame size, we're probably only allocating as much as necessary anyway, so this helps us again remove the guard in `stack_get` to check for None